### PR TITLE
Remove lint warnings

### PR DESCRIPTION
**Pull request by hannesa2**
I followed the suggestion from lint

### CODE REVIEW

File: tracker/src/main/java/org/matomo/sdk/TrackMe.java [hunk 1]
```diff
@@ -94,37 +93,33 @@ public synchronized boolean has(@NonNull QueryParams queryParams) {
      *
      * @param key   type
      * @param value value
-     * @return this (for chaining)
      */
-    public synchronized TrackMe trySet(@NonNull QueryParams key, int value) {
-        return trySet(key, String.valueOf(value));
+    public synchronized void trySet(@NonNull QueryParams key, int value) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackMe.java:97]: Changes public API, see above.

File: tracker/src/main/java/org/matomo/sdk/TrackMe.java [hunk 2]
```diff
@@ -94,37 +93,33 @@ public synchronized boolean has(@NonNull QueryParams queryParams) {
...
     /**
      * Only sets the value if it doesn't exist.
      *
      * @param key   type
      * @param value value
-     * @return this (for chaining)
      */
-    public synchronized TrackMe trySet(@NonNull QueryParams key, float value) {
-        return trySet(key, String.valueOf(value));
+    public synchronized void trySet(@NonNull QueryParams key, float value) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackMe.java:107]: Changes public API, see above.

File: tracker/src/main/java/org/matomo/sdk/TrackMe.java [hunk 3]
```diff
@@ -94,37 +93,33 @@ public synchronized boolean has(@NonNull QueryParams queryParams) {
...
+        trySet(key, String.valueOf(value));
     }
 
-    public synchronized TrackMe trySet(@NonNull QueryParams key, long value) {
-        return trySet(key, String.valueOf(value));
+    public synchronized void trySet(@NonNull QueryParams key, long value) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackMe.java:111]: Changes public API, see above.

File: tracker/src/main/java/org/matomo/sdk/TrackMe.java [hunk 4]
```diff
@@ -94,37 +93,33 @@ public synchronized boolean has(@NonNull QueryParams queryParams) {
...
 
     /**
      * Only sets the value if it doesn't exist.
      *
      * @param key   type
      * @param value value
-     * @return this (for chaining)
      */
-    public synchronized TrackMe trySet(@NonNull QueryParams key, String value) {
+    public synchronized void trySet(@NonNull QueryParams key, String value) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackMe.java:121]: Changes public API, see above.

File: tracker/src/main/java/org/matomo/sdk/TrackerBuilder.java [hunk 5]
```diff
@@ -43,9 +43,8 @@ public int getSiteId() {
     /**
      * A unique name for this Tracker. Used to store Tracker settings independent of URL and id changes.
      */
-    public TrackerBuilder setTrackerName(String name) {
+    public void setTrackerName(String name) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackerBuilder.java:46]: Changes public API (chaining calls on a builder object `Builder().doA().doB()`

File: tracker/src/main/java/org/matomo/sdk/TrackerBuilder.java [hunk 6]
```diff
@@ -58,9 +57,8 @@ public String getTrackerName() {
      *
      * @param domain your-domain.com
      */
-    public TrackerBuilder setApplicationBaseUrl(String domain) {
+    public void setApplicationBaseUrl(String domain) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/TrackerBuilder.java:60]: Changes public API

File: tracker/src/main/java/org/matomo/sdk/extra/CustomVariables.java [hunk 7]
```diff
@@ -70,9 +70,8 @@ public CustomVariables(@Nullable String json) {
         }
     }
 
-    public CustomVariables putAll(CustomVariables customVariables) {
+    public void putAll(CustomVariables customVariables) {
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/extra/CustomVariables.java:73]: Changes public API

File: tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java [hunk 8]
```diff
@@ -88,12 +85,6 @@ public String buildExtraIdentifier() {
             }
         }
 
-        /**
-         * Custom exta identifier. Supply your own \o/.
-         */
-        abstract class Custom implements Extra {
-        }
-
```
Comment (d4rken) [inline tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java:96]: I agree, this is kind of superfluos, but it's part of the public API and some devs might have extended this.

**Review by d4rken — CHANGES_REQUESTED** [review]
Some of the changes affect the public API that devs might be using (see comment), the rest looks good, thanks for the cleanup :+1:

File: tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java (hunk 8 above)
Comment (hannesa2) [reply tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java:96]: Keeping this unused stuff is a kind of collect a lot waste in your base ground.

File: tracker/src/main/java/org/matomo/sdk/Matomo.java [hunk 9]
```diff
@@ -30,7 +30,7 @@ public class Matomo {
     private static final String TAG = Matomo.tag(Matomo.class);
     private static final String BASE_PREFERENCE_FILE = "org.matomo.sdk";
 
-    @SuppressLint("StaticFieldLeak") private static Matomo sInstance;
+    @SuppressLint("StaticFieldLeak") private static volatile Matomo sInstance;
```
[the remark this answers is not in the document]
Comment (hannesa2) [reply tracker/src/main/java/org/matomo/sdk/Matomo.java:33]: image

File: tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java (hunk 8 above)
Comment (d4rken) [reply tracker/src/main/java/org/matomo/sdk/extra/DownloadTracker.java:96]: We can remove it, but then with a major version bump and a little note about migration in the changelog (e.g. don't extend `Custom`, just implement `Extra`).

File: tracker/src/main/java/org/matomo/sdk/Matomo.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (d4rken) [reply tracker/src/main/java/org/matomo/sdk/Matomo.java:33]: Visibility issue when accessing it from multiple threads should be handled by `synchronized` already :thinking: 

Hm, it doesn't hurt.  :shrug:

**Review by d4rken — APPROVED** [review]
Thank you!

File: tracker/src/main/java/org/matomo/sdk/TrackMe.java [hunk 10]
```diff
@@ -30,9 +30,8 @@ public TrackMe(TrackMe trackMe) {
     /**
      * Adds TrackMe to this TrackMe, overriding values if necessary.
      */
-    public TrackMe putAll(@NonNull TrackMe trackMe) {
+    public void putAll(@NonNull TrackMe trackMe) {
```
[the remark this answers is not in the document]
Comment (hannesa2) [reply tracker/src/main/java/org/matomo/sdk/TrackMe.java]: Sure, I as a customer will only use stuff I find in sample app. It's just a note.

> The linter doesn't know it's an SDK and code is used outside the project

yes, maybe annotate it with "ignore warning" is a way to go

[the remark this answers is not in the document]
Comment (hannesa2) [reply tracker/src/main/java/org/matomo/sdk/TrackMe.java]: Till now, I ignored my wish to kotlize it.
Kotlin has some wonderful mechanism for a sdk, e.g `internal`

**Comment by hannesa2** [comment]
Nice, we come closer 
image

**Comment by hannesa2** [comment]
But this interface way isn't done

image

**Comment by d4rken** [comment]
> This keeping unused classes because someone could use it, is an afford without benefit.

Why do you keep coming back to this? :smile: 
I know it's not great, I probably just forgot about that class. I had no one to review my code.
It's out there now and we shouldn't break the users app "just because it's not clean".
Sorry for not writing the perfect SDK? :thinking: 

> But this interface way isn't done

Not everything needs to be a firebase-level SDK with shovels of abstraction.
This is volunteer work and the SDK worked fine for my use-cases, so at that time: Why make things more complicated if there is no requirement?
A rewrite to Kotlin wasn't on the horizon at that point.

Interfaces can still be introduced now if desired. As said, you can make a v5, with everything new and better if you want to  :+1:.

**Comment by hannesa2** [comment]
Sorry, those were just a few thoughts on the Kotlin transformation.
You are right, house holding of our time is also important.

**Pull request merged**